src/libsyntax/parse/lexer/tokentrees.rs

```diff
@@ -223,8 +223,35 @@ impl<'a> TokenTreesReader<'a> {
                 _ => {
                     self.token = token;
                     return;
-                },
+                }
+            }
+        }
+    }
+}
+
+struct TokenStreamBuilder {
```

Comment:
```suggestion
#[derive(Default)]
struct TokenStreamBuilder {
```

Reply:
Seems like a strange argument to me to intentionally not take advantage of the language's powerful facilities to get rid of the boilerplate where possible. I would think that especially in one-off local cases, deriving is a good idea.

Reply:
I feel like this is a place where we have different values: for me, avoiding powerful facilities is worth it, if the cost is just a bit of a boilerplate :)

However, I value not arguing over small stuff even more :)